FIX #19064 - access forbidden when editing other user's password

atm-florianm opened the pull request:
# FIX #19064 
When a user does not have permission to create other users but has permission to edit other users' passwords, when they do so, they land on a rejection page.

Since this fix touches core security features, I wouldn't mind extra scrutiny to make sure it doesn't introduce any new vulnerability.

atm-florianm committed:
FIX #19064 - access forbidden when editing other user's password despite relevant permissions OK

eldy commented:
What happen after your fix if:
User has only permission to edit password of others and user try to edit the firstname ?

atm-florianm commented:
> What happen after your fix if: User has only permission to edit password of others and user try to edit the firstname ?

I just tested this scenario:
1) (as admin) I gave user A the right to edit other users
2) (as user A) I opened the edit form for user B and changed the first name
3) (as admin) I removed permission from user A
4) (as user A) I submitted the form

Result:
No forbidden access, but also no first name change for user B (the modification was not saved, which is what we expect with respect to permissions).

atm-florianm commented:
Closing the PR: I realize my PR is a duplicate of #19065 (though fixes differ slightly).
